components/nexus-core/src/test/java/org/sonatype/nexus/internal/app/BaseUrlManagerImplTest.java

```diff
@@ -0,0 +1,85 @@
+package org.sonatype.nexus.internal.app;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.sonatype.sisu.litmus.testsupport.TestSupport;
+
+import org.mockito.InjectMocks;
+
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.mockito.Mockito.when;
+import com.google.inject.Provider;
+import org.junit.Test;
+import org.mockito.Mock;
+import static org.junit.Assert.assertThat;
+
+/**
+ * Tests for {@link BaseUrlManagerImpl}
+ */
+public class BaseUrlManagerImplTest
+    extends TestSupport
+{
+  static final String NUGET_QUERY = "/repository/nuget.org-proxy/Packages(Id='jQuery',Version='2.1.4')";
+  static final String ENCODED_NUGET_QUERY = "/repository/nuget.org-proxy/Packages(Id=%27jQuery%27,Version=%272.1.4%27)";
+  static final String FAKE_URL = "http://example.com:1234/foo/bar";
+
+  @Mock
+  Provider<HttpServletRequest> requestProvider;
+
+  @Mock
+  HttpServletRequest request;
+
+  @InjectMocks
+  BaseUrlManagerImpl underTest;
```

Comment:
why? we're avoiding framework injection and utilizing a built-in feature of mockito so I don't understand your objection here